### ISSUE TITLE
New compiler: code lines that contain pre-evaluated math are missed during the step-by-step execution

### ISSUE THREAD

**Issue by ivan-mogilko** [adventuregamestudio/ags#1832]
**Describe the bug**
The new compiler in ags4 branch can precalculate simple arithmetical operations. So, if you have a variable declaration which assignment includes simple math (e.g. `int a = 2 + 2;`), the resulting value will be evaluated and used when writing a bytecode, instead of the sequence of operators.
It seems that when such line is written to the bytecode, it omits some information, like a line start opcode. Because of that, when running the game under debugger, then breakpoint on this line will never trigger, and when doing step-by-step execution, this line will be missed. This is inconsistent with the simpler assignments, such as `int b = 4`;

**AGS Version**
AGS 4

**To Reproduce**
Use following script:
```
function game_start()
{
   int a = 1;
   int b = 1 + 1;
   int c = a + b;
   c += 1;
   c += 1 + 1;
   return 0;
}
```

In General Settings make sure that the "Use extended script compiler" is enabled.
Place a breakpoint at the first line (`int a = 1;`), then run the game under debugger. When the breakpoint is hit, press F11 to make a step. Notice that the debugger will only stop on following lines, jumping the ones between which contain pre-evaluated math:
```
   int a = 1;
   int b = 1 + 1; // <---- skipped
   int c = a + b;
   c += 1;
   c += 1 + 1; // <---- skipped
   return 0;
```

**Expected behavior**
Script lines should not be missed by the debugger when doing step-by-step execution.

**Comment by ericoporto** [comment]
Uhm, do you know if for `int a = 1; int b = 1 + 1; int c = b + 1;` the compiler is optimizing for `int c = 3` ?

**Comment by ivan-mogilko** [comment]
> Uhm, do you know if for `int a = 1; int b = 1 + 1; int c = b + 1;` the compiler is optimizing for `int c = 3` ?

It does not, like I mentioned, the debugger stops on first (`int a = 1`) and third (`int c = b + 1`) lines.

**Comment by ivan-mogilko** [comment]
> Uhm, do you know if for `int a = 1; int b = 1 + 1; int c = b + 1;` the compiler is optimizing for `int c = 3` ?

It does not look like it does. For a better test I used this script:
```
  int a = 1;
  int b = 1 + 1;
  int c = a + b;
```

The debugger stops on the first and third line, skipping the second. If it was optimizing, it would only stop on the third.

**Comment by fernewelten** [comment]
I'll have a look. Most of the bytecode googletests that we have so far run with line numbers switched off, so it might be that the compiler is forgetting a linenum directive somewhere that the googletests haven't caught.

**Comment by fernewelten** [comment]
Addressed in #1835.

**Comment by fernewelten** [comment]
> Most of the bytecode googletests that we have so far run with line numbers switched off

Looking back at this decision,  IMHO this is a bad one. In those tests where we're checking just what bytes are being generated, we absolutely should also test just what and just where `LINENUM` pseudo-commands are generated. 

But this is separate from the current bug, so I'll post an issue for it.
